apps/src/pythonlab/pyodideWorkerManager.ts

```
@@ -93,24 +104,86 @@ const setUpPyodideWorker = () => {
   return worker;
 };
 
+const canSupportInput = () => {
+  // We can support input if service workers are supported by the current browser
+  // and the python input experiment is enabled.
+  return (
+    'serviceWorker' in navigator &&
+    experiments.isEnabled(experiments.PYTHON_INPUT)
+  );
+};
+
+const registerServiceWorker = async () => {
+  // No-op if service workers are not supported.
+  if (canSupportInput()) {
+    try {
+      const url = new URL(
+        './inputServiceWorker.js',
+        // @ts-expect-error because TypeScript does not like this syntax.
+        import.meta.url
+      );
+      const registration = await navigator.serviceWorker.register(url);
+      if (registration.active) {
+        inputServiceWorker = registration.active;
+      }
+
+      registration.addEventListener('updatefound', () => {
+        const installingWorker = registration.installing;
+        if (installingWorker) {
+          installingWorker.addEventListener('statechange', () => {
+            if (installingWorker.state === 'installed') {
+              inputServiceWorker = installingWorker;
+            }
+          });
+        }
+      });
+    } catch (error) {
+      console.error(`Registration failed with ${error}`);
```

Probably a future PR question, but: "What _should_ we do once curriculum depends on this if this fails?" Potentially we want to let the exception bubble up because won't it make it to new relic then?

---

Yeah we should handle this with more than a console error. I'll file a follow-up task.